caddyhttp/proxy/proxy.go

```diff
@@ -173,15 +192,20 @@ func (p Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error) {
 			downHeaderUpdateFn = createRespHeaderUpdateFn(host.DownstreamHeaders, replacer)
 		}
 
-		// rewind request body to its beginning
-		if err := body.rewind(); err != nil {
-			return http.StatusInternalServerError, errors.New("unable to rewind downstream request body")
+		// NOTE: We check for requiresBuffering, because
+		// only a bufferedBody supports rewinding.
+		if requiresBuffering {
```

Comment:
I think we could clear this checking if `unbufferBody`'s rewind() just return nil